src/clj/puppetlabs/services/jruby_pool_manager/impl/reference_pool.clj

```diff
@@ -31,7 +39,44 @@
       (.unlock pool)))
 
   (borrow
-    [this])
+    [this]
+    (let [pool (jruby-internal/get-pool this)]
+      (.borrowItem pool)))
 
   (return
-    [this instance]))
+    [this instance]
+    (let [pool (jruby-internal/get-pool this)
+          borrow-count (:borrow-count this)
+          max-borrows (get-in instance [:internal :max-borrows])
+          modify-instance-agent (jruby-agents/get-modify-instance-agent this)]
+      (.releaseItem pool instance)
+      (swap! borrow-count inc)
```

Comment:
perhaps there's a less-overloaded / clearer term than `borrow-count` here. maybe like `total-borrows-before-flush` or something like that..

Reply:
or maybe `total-requests-per-instance`, similar to `max-requests-per-instance` like the commit says